src/MovedDiscussionFirstPostFactory.php

```diff
@@ -55,6 +55,10 @@ public function create(Discussion $sourceDiscussion, Discussion $targetDiscussio
 
         $newFirstPost->save();
 
+        // save number again, since it is overwritten by `Post::creating()` hook
+        $newFirstPost->number = $oldFirstPost->number;
+        $newFirstPost->save();
+
```

Comment:
Fo the long run it might be good to actually save silently to avoid anything like that listener in core making unwanted change since this is an edge case we probably wouldn't want such changes.
```suggestion
        $newFirstPost->saveQuietly();
```